VSocks.md

```diff
@@ -0,0 +1,133 @@
+
+- [Introduction](#introduction)
+    - [proxy communication diagram](#proxy-communication-diagram)
+    - [vsock communication diagram](#vsock-communication-diagram)
+- [System requirements](#system-requirements)
+- [Advantages of using vsocks](#advantages-of-using-vsocks)
+    - [High density](#high-density)
+    - [Reliability](#reliability)
+
+## Introduction
+
+There are two different ways processes in the virtual machine can communicate
+with processes in the host. The first one is by using serial ports, where the
+processes in the virtual machine can read/write data from/to a serial port device
+and the processes in the host can read/write data from/to a unix socket. This
+solution has been part of QEMU from long time ago and almost all GNU/Linux
+distributions have support for serial ports, making it the most portable solution.
+However, there is a limitation that impact its usage and implementation. The
+communication between processes is serial, making impossible that two or more
+processes read/write data at the same time. To deal with this limitation the
+resources (serial port and unix socket) must be multiplexed. In Kata Containers
+those resources are multiplexed by using [kata-proxy][2] and [yamux][3], the
+following diagram shows how it's implemented.
+
+
+### proxy communication diagram
+
+```
+.----------------------.
+| .------------------. |
+| | .-----.  .-----. | |
+| | |cont1|  |cont2| | |
+| | `-----'  `-----' | |
+| |       \   /      | |
+| |    .---------.   | |
+| |    |  agent  |   | |
+| |    `---------'   | |
+| |         |        | |
+| |    .-----------. | |
+| |POD |serial port| | |
+| `----|-----------|-' |
+|      |  socket   |   |
+|      `-----------'   |
+|           |          |
+|       .-------.      |
+|       | proxy |      |
+|       `-------'      |
+|           |          |
+|  .------./ \.------. |
+|  | shim |   | shim | |
+|  `------'   `------' |
+| Host                 |
+`----------------------'
+```
+
+The second way for processes to communicate, and the newer one, is by using
+[vsocks][4]. The communication through vsocks is not serial and can accept
+connections from multiple clients unlike serial ports devices, therefore no
+multiplexers ([kata-proxy][2] and [yamux][3]) are required. The following diagram
+shows how it's implemented in Kata Containers.
+
+
+### vsock communication diagram
+
+```
+.----------------------.
+| .------------------. |
+| | .-----.  .-----. | |
+| | |cont1|  |cont2| | |
+| | `-----'  `-----' | |
+| |       |   |      | |
+| |    .---------.   | |
+| |    |  agent  |   | |
+| |    `---------'   | |
+| |       |   |      | |
+| | POD .-------.    | |
+| `-----| vsock |----' |
+|       `-------'      |
+|         |   |        |
+|  .------.   .------. |
+|  | shim |   | shim | |
+|  `------'   `------' |
+| Host                 |
+`----------------------'
+```
+
+## System requirements
+
+Host linux kernel version must be greater than or equal to v4.8 and the
```

Comment:
Lines 90 and 91 - "The host Linux kernel version must be greater than or equal to v4.8, and ..."